### PR TITLE
Properly attach inner attributes in Attrs::new

### LINKED ISSUE #2148: Inner comments are attached to the wrong things
Currently the following are equivalent:

1.
```rust
//! A comment
mod foo {}
```

and

2. 
```rust
/// A comment
mod foo {}
```

I believe the first example is incorrect and the comment should apply to the parent scope.

The following should be equivalent to 2 above but the comment here is attached to the item list and not the module:

```rust
mod foo {
  //! A comment
}
```

### PR DESCRIPTION
**Pull request by Veykril**
Properly attach inner and outer attributes to the things they actually belong to in the HIR. ~~I can add some tests for this if wanted once I know where to put them/how to test for this.~~ Put some tests into `hover.rs`.
 
So the following snippet
```rust
mod foo {
	//! Hello
}
```
now shows `Hello` on hover 🎉

Fixes #2148

### CODE REVIEW

**Comment by kjeremy** [comment]
THIS. This makes me happy.

File: crates/hir_def/src/attr.rs [hunk 1]
```diff
@@ -184,6 +191,41 @@ impl Attrs {
...
+                (assoc_item_list.attrs(), ast::CommentIter::from_syntax_node(assoc_item_list.syntax()))
+            },
+            ast::Module(it) => {
+                let item_list = it.item_list()?;
+                (item_list.attrs(), ast::CommentIter::from_syntax_node(item_list.syntax()))
+            },
+            // FIXME: BlockExpr's only accept inner attributes in specific cases
+            // Excerpt from the reference:
+                // Block expressions accept outer and inner attributes, but only when they are the outer
+                // expression of an expression statement or the final expression of another block expression.
```
Comment (kjeremy) [inline crates/hir_def/src/attr.rs:219]: Formatting

**Comment by kjeremy** [comment]
While it's not scoped to this specific spot in the code you could put some tests in hover or somewhere else where we render doc comments so we know that we don't break anything there going forward.

**Comment by Veykril** [comment]
That doesn't seem to be a bad place given that also allows testing attributes by using the attribute form of docs.

**Pull request merged**